kernel/src/hil/hw_debug.rs

```diff
@@ -0,0 +1,31 @@
+//! Interfaces for interacting with debug hardware integrated in various SoCs.
+//! Currently allows reading the cycle counter.
+use crate::ErrorCode;
+
+pub trait CycleCounter {
+    /// Enable and start the cycle counter.
+    /// Returns NOSUPPORT if the cycle counter is not present.
+    fn start(&self) -> Result<(), ErrorCode>;
```

Comment:
```suggestion
    /// Enable and start the cycle counter.
    fn start(&self);
```

Reply:
I went ahead and changed this, but I think that means if someone tries to use this on a chip with a DCB peripheral but no cycle counter it may fail silently. I think that is fine though.